internal/executor/job/job_context.go

```diff
@@ -287,3 +303,14 @@ func (c *ClusterJobContext) handleDeletedPod(pod *v1.Pod) {
 		}
 	}
 }
+
+func (c *ClusterJobContext) checkForPreemptionEvent(pod *v1.Pod) *v1.Event {
+	events := c.clusterContext.GetEvents()
```

Comment:
Are we able to use an indexer here as in https://github.com/G-Research/armada/pull/1504/files#diff-67e3b00cf2b0b449d3dc3bee9dbf9b7ec3011c458b9bf79527dabdbecafa96bfR208

Reply:
Yes, good catch